src/ts/state/reducer.ts

```diff
@@ -146,25 +154,29 @@ export const PROSEMIRROR_TYPERIGHTER_ACTION = "PROSEMIRROR_TYPERIGHTER_ACTION";
 export const createInitialState = <TMatch extends IMatch>(
   doc: Node,
   matches: TMatch[] = [],
-  active: boolean = true
-): IPluginState<TMatch> => ({
-  config: {
-    isActive: active,
-    debug: false,
-    requestMatchesOnDocModified: false
-  },
-  decorations: DecorationSet.create(doc, createDecorationsForMatches(matches)),
-  dirtiedRanges: [],
-  currentMatches: matches,
-  selectedMatch: undefined,
-  hoverId: undefined,
-  hoverInfo: undefined,
-  requestsInFlight: {},
-  requestPending: false,
-  requestErrors: []
-});
-
-export const createReducer = (expandRanges: ExpandRanges) => {
+  active: boolean = true,
+  ignoreMatch: IIgnoreMatch = () => true
+): IPluginState<TMatch> => {
+  const initialState: IPluginState<TMatch> = {
+    config: {
+      isActive: active,
+      debug: false,
+      requestMatchesOnDocModified: false
+    },
+    decorations: DecorationSet.create(doc, []),
+    dirtiedRanges: [],
+    currentMatches: [],
+    selectedMatch: undefined,
+    hoverId: undefined,
+    hoverInfo: undefined,
+    requestsInFlight: {},
+    requestPending: false,
+    requestErrors: []
+  };
+  return addMatchesToState(initialState, doc, matches, ignoreMatch);
```

Comment:
The consumer can add matches when the plugin initialises – we need to run those matches through the `ignoreMatch` predicate, so a new helper function, `addMatchesToState`, takes care of that task.